### PR TITLE
SetupViewController with Texture

### PR DESCRIPTION
**Pull request by Kharchevskyi**
this was for issue #58

### CODE REVIEW

**Comment by Kharchevskyi** [comment]
Simulator Screen Shot - iPhone Xs Max - 2019-10-29 at 20 19 57

File: FlowCrypt/Controllers/Setup/SetupViewController.swift [hunk 1]
```diff
@@ -2,39 +2,71 @@
...
-final class SetupViewController: UIViewController {
-    private enum Constants {
-        static let noBackups = "No backups found on account: \n"
-        static let actionFailed = "Action failed"
-        static let useOtherAccount = "Use other account"
-        static let enterPassPhrase = "Enter pass phrase"
-        static let wrongPassPhraseRetry = "Wrong pass phrase, please try again"
+final class SetupViewController: ASViewController<ASTableNode> {
+    private enum Parts: Int, CaseIterable {
+        case title, description, passPrase, divider, action, optionalAction
```
Comment (tomholub) [inline FlowCrypt/Controllers/Setup/SetupViewController.swift]: typo `passPrase`

File: FlowCrypt/Controllers/Setup/SetupViewController.swift [hunk 2]
```diff
@@ -87,34 +112,39 @@ extension SetupViewController {
...
-            }
+            self?.handleSuccessfullBackUp()
         }.catch(on: .main) { [weak self] error in
-            self?.renderNoBackupsFoundOptions(msg: Constants.actionFailed, error: error)
+            self?.renderNoBackupsFoundOptions("setup_action_failed".localized, error: error)
         }
     }
 
-    private func renderNoBackupsFoundOptions(msg: String, error: Error? = nil) {
+    private func handleSuccessfullBackUp() {
```
Comment (tomholub) [inline FlowCrypt/Controllers/Setup/SetupViewController.swift]: rename to `handleBackupsFetchResult`

File: FlowCrypt/Controllers/Setup/SetupViewController.swift [hunk 3]
```diff
@@ -128,38 +158,19 @@ extension SetupViewController {
...
-        }
-    }
 
     private func recoverAccountWithBackups(with passPhrase: String) {
         let matchingBackups: [KeyDetails] = fetchedEncryptedPrvs
             .compactMap { (key) -> KeyDetails? in
-                guard let prv = key.private else { return nil }
-                guard let r = try? Core.decryptKey(armoredPrv: prv, passphrase: passPhrase), r.decryptedKey != nil else { return nil }
+                guard let privateKey = key.private,
+                    let recovery = try? Core.decryptKey(armoredPrv: privateKey, passphrase: passPhrase),
```
Comment (tomholub) [inline FlowCrypt/Controllers/Setup/SetupViewController.swift]: rename to `let decrypted`

File: FlowCrypt/Controllers/Setup/SetupViewController.swift [hunk 4]
```diff
@@ -230,18 +237,73 @@ extension SetupViewController {
...
+    func tableNode(_ tableNode: ASTableNode, nodeBlockForRowAt indexPath: IndexPath) -> ASCellNodeBlock {
+        return { [weak self] in
+            guard let self = self, let part = Parts(rawValue: indexPath.row) else { return ASCellNode() }
+            switch part {
+            case .title:
+                return SetupTitleNode(SetupStyle.title, insets: SetupStyle.titleInset)
+            case .description:
+                return SetupTitleNode(SetupStyle.subtitleStyle(self.subtitle), insets: SetupStyle.subTitleInset)
+            case .passPrase:
+                return SetupPassPraseNode() { [weak self] value in
```
Comment (tomholub) [inline FlowCrypt/Controllers/Setup/SetupViewController.swift]: Typo `SetupPassPraseNode`

**Pull request merged**